File: Algorithms/1342. Number of Steps to Reduce a Number to Zero/python/solution.bitwise.py
class Solution:
    def numberOfSteps(self, num: int) -> int:
        # We need to handle this as a special case, otherwise it'll return -1.
        if num == 0:
            return 0

        steps: int = 0
        power_of_two: int = 1

        while power_of_two <= num:
            # Apply the bit mask to check if the bit at "power_of_two" is a 1.
            if (power_of_two & num) != 0:
                steps += 2
            else:
                steps += 1
            power_of_two *= 2

        # We need to subtract 1, because the last bit was over-counted.
        return steps - 1

File: Algorithms/1342. Number of Steps to Reduce a Number to Zero/kotlin/solution.bitwise.kt
class Solution {
    fun numberOfSteps(num: Int): Int {
        if (num == 0) {
            return 0
        }

        var steps: Int = 0
        var powerOfTwo: Int = 1
        while (powerOfTwo <= num) {
            if (num and powerOfTwo != 0) {
                steps += 2
            } else {
                steps += 1
            }
            powerOfTwo *= 2
        }
        return steps - 1
    }
}
